packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -831,21 +830,25 @@ function renderRoot(
   } else if (workInProgressRootExitStatus === RootSuspendedWithDelay) {
     // We could've received an update at a lower priority while we yielded.
     // We're suspended in a delayed state. Once we complete this render we're
-    // just going to try to recover at the last pending time anyway so we might
-    // as well start doing that eagerly.
+    // just going to try to recover at the pending time anyway so we might as
+    // well start doing that eagerly.
+    //
     // Ideally we should be able to do this even for retries but we don't yet
     // know if we're going to process an update which wants to commit earlier,
     // and this path happens very early so it would happen too often. Instead,
     // for that case, we'll wait until we complete.
     if (workInProgressRootHasPendingPing) {
       // We have a ping at this expiration. Let's restart to see if we get unblocked.
       prepareFreshStack(root, expirationTime);
-    } else {
-      const lastPendingTime = root.lastPendingTime;
-      if (lastPendingTime < expirationTime) {
-        // There's lower priority work. It might be unsuspended. Try rendering
-        // at that level immediately, while preserving the position in the queue.
-        return renderRoot.bind(null, root, lastPendingTime);
```

Comment:
We're discussing this.  Seems like this refactoring changed things.

Reply:
This can be replaced by a prepareFreshStack in an update that happens during a yield.

If the current tree has already determined that it has remaining work at lower pri, we can use that info to eagerly throw to terminate the render and try at lower pri.